src/net/sourceforge/kolmafia/textui/function/LibraryClassFunction.java

```diff
@@ -0,0 +1,61 @@
+package net.sourceforge.kolmafia.textui.function;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.ArrayList;
+import java.util.List;
+import net.sourceforge.kolmafia.textui.DataTypes;
+import net.sourceforge.kolmafia.textui.parsetree.Function;
+import net.sourceforge.kolmafia.textui.parsetree.LibraryFunction;
+import net.sourceforge.kolmafia.textui.parsetree.Type;
+
+public abstract class LibraryClassFunction {
+  String functionName;
+
+  LibraryClassFunction(final String functionName) {
+    this.functionName = functionName;
+  }
+
+  @Retention(RetentionPolicy.RUNTIME)
+  @Target({ElementType.METHOD})
+  public @interface LibraryFunctionOverload {
+    String returns() default "void";
+  }
+
+  @Retention(RetentionPolicy.RUNTIME)
+  @Target({ElementType.PARAMETER})
+  public @interface LibraryFunctionParameter {
+    String value();
+  }
+
+  public final List<Function> getFunctions() {
```

Comment:
Perhaps this should return a `FunctionList`? I'm not sure what the benefit would be.

Reply:
Well, I plan to make it so that function references are saved in TypeList, VariableList and FunctionList, so there's that... no sure if it's relevant here, though...